ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,932 @@
+        ? [this.poolState.tokenAMint, this.poolState.tokenBMint]
+        : [this.poolState.tokenBMint, this.poolState.tokenAMint];
+
+    const adminTokenFee =
+      this.tokenA.address === inTokenMint.toBase58() ? this.poolState.adminTokenAFee : this.poolState.adminTokenBFee;
+
+    let preInstructions: Array<TransactionInstruction> = [];
+    const [[userSourceToken, createUserSourceIx], [userDestinationToken, createUserDestinationIx]] =
+      await this.createATAPreInstructions(owner, [sourceToken, destinationToken]);
+
+    createUserSourceIx && preInstructions.push(createUserSourceIx);
+    createUserDestinationIx && preInstructions.push(createUserDestinationIx);
+
+    if (sourceToken.equals(WRAPPED_SOL_MINT)) {
+      preInstructions = preInstructions.concat(wrapSOLInstruction(owner, userSourceToken, inAmountLamport.toNumber()));
+    }
+
+    preInstructions.push(await this.getApySyncInstructions());
+
+    const postInstructions: Array<TransactionInstruction> = [];
+    if (WRAPPED_SOL_MINT.equals(destinationToken)) {
+      const unwrapSOLIx = await unwrapSOLInstruction(owner);
+      unwrapSOLIx && postInstructions.push(unwrapSOLIx);
+    }
+
+    const swapTx = await this.program.methods
+      .swap(inAmountLamport, outAmountLamport)
+      .accounts({
+        aTokenVault: this.vaultA.vaultState.tokenVault,
+        bTokenVault: this.vaultB.vaultState.tokenVault,
+        aVault: this.poolState.aVault,
+        bVault: this.poolState.bVault,
+        aVaultLp: this.poolState.aVaultLp,
+        bVaultLp: this.poolState.bVaultLp,
+        aVaultLpMint: this.vaultA.vaultState.lpMint,
+        bVaultLpMint: this.vaultB.vaultState.lpMint,
+        userSourceToken,
+        userDestinationToken,
+        user: owner,
+        adminTokenFee,
+        pool: this.address,
+        tokenProgram: TOKEN_PROGRAM_ID,
+        vaultProgram: this.vaultProgram.programId,
+      })
+      .remainingAccounts(getRemainingAccounts(this.poolState))
+      .preInstructions(preInstructions)
+      .postInstructions(postInstructions)
+      .transaction();
+
+    return new Transaction({
+      feePayer: owner,
+      ...(await this.program.provider.connection.getLatestBlockhash('finalized')),
+    }).add(swapTx);
+  }
+
+  /**
+   * `getDepositQuote` is a function that takes in a tokenAInAmount, tokenBInAmount, balance, and
+   * slippage, and returns a poolTokenAmountOut, tokenAInAmount, and tokenBInAmount
+   * @param {BN} tokenAInAmount - BN,
+   * @param {BN} tokenBInAmount - BN,
+   * @param {boolean} [balance=true] - return false if the deposit is imbalance (default: true)
+   * @param {number} [slippage] - The amount of slippage you're willing to accept.
+   * @returns The return value is a tuple of the poolTokenAmountOut, tokenAInAmount, and
+   * tokenBInAmount.
+   */
+  public async getDepositQuote(
+    tokenAInAmount: BN,
+    tokenBInAmount: BN,
+    balance = true,
+    slippage?: number,
```

Comment:
I think we should not default initialize the param, and enforce them to pass in 
balancedDeposit, and slippage